lambda/github_api.py

```diff
@@ -0,0 +1,97 @@
+import os
+import time
+
+import requests
+
+import config
+from logger import LOG
+
+TOKEN_PREFIX = "/alert-processor/tokens/github/"
+API_ROOT = "https://api.github.com"
+ACCESS_TOKEN = None
+
+
+def set_github_access_token(token):
+    """ Get the access token once """
+    global ACCESS_TOKEN
+    ACCESS_TOKEN = token
+
+
+def get_github_access_token():
+    """ Get the access token once """
+    global ACCESS_TOKEN
+    if ACCESS_TOKEN is None:
+        access_token_param = os.environ.get("GITHUB_TOKEN")
+        ACCESS_TOKEN = config.get_ssm_param(access_token_param)
+    return ACCESS_TOKEN
+
+
+def get_github_api_paged_data(url: str) -> list:
+    token = get_github_access_token()
+    page = 1
+    page_size = 100
+    page_items = 1
+    items = []
+    while page_items > 0:
+        page_items = 0
+        page_url = f"{url}?page={page}&per_page={page_size}"
+        headers = {"authorization": f"token {token}"}
+        response = requests.get(page_url, headers=headers).json()
```

Comment:
If a single request fails it will break the whole run. Do you need some retry logic for this?

Reply:
It'll break at that branching point yes. I think we probably need to catch errors (especially rate-limit errors) and flag up if we start getting incomplete runs.

Reply:
This can be moved to a new issue.

Reply:
I've raised #15 for this.